AlexanderOtavka/ride-board
# Notify per ride

Resolves #22 

# Changes

- Notifications have been decoupled from being in the ride
  - You can receive notifications for a ride you are not in
  - You can turn off notifications for a ride you are in
- Posting a message in a ride automatically turns on notifications for that ride
- There is a new root notifications page where you can change the user-based preferences

# Manual Testing

Create an initial test user with your real Grinnell email so you will actually receive email notifications.

For both passenger and driver apps:
1. Play with user-level notifications
  1. Go to notification preferences from the rides page
  2. Turn on email notifications and text message notifications, you should receive an email and a text confirming
  3. Turn off text notifications to save money (email notifications are free)
2. Play with ride-level notifications
  1. Create a ride
    - You should be subscribed for notifications automatically
  2. In an incognito window, create a separate user with a fake email
  3. Post a message as the other user
    - This should automatically subscribe that user
    - You should receive an email about the message posted
  4. Join the ride as the other user, then leave
    - You should receive an email about them joining
    - And another about them leaving
  5. Unsubscribe yourself from that ride's notifications
  6. Send another message from the fake user
    - You should **not** receive an email
  7. Join and leave the ride again as the other user
    - You should receive no emails about this either
  8. Create a ride as the other user
  9. Post a message on the other user's ride
    - This should subscribe you to the ride
  10. Respond to that message as the other user
    - You should receive an email
